c/driver/postgresql/statement.cc

```diff
@@ -500,33 +540,48 @@ struct BindStream {
               std::memcpy(param_values[col] + sizeof(uint64_t), &days, sizeof(uint32_t));
               std::memcpy(param_values[col] + sizeof(uint64_t) + sizeof(uint32_t),
                           &months, sizeof(uint32_t));
+              field_nbytes = 16;
               break;
             }
             default:
               SetError(error, "%s%" PRId64 "%s%s%s%s", "[libpq] Field #", col + 1, " ('",
                        bind_schema->children[col]->name,
                        "') has unsupported type for ingestion ",
                        ArrowTypeString(bind_schema_fields[col].type));
+              ArrowBufferReset(&buf);
               return ADBC_STATUS_NOT_IMPLEMENTED;
           }
-        }
 
-        result = PQexecPrepared(conn, /*stmtName=*/"",
-                                /*nParams=*/bind_schema->n_children, param_values.data(),
-                                param_lengths.data(), param_formats.data(),
-                                /*resultFormat=*/0 /*text*/);
-
-        ExecStatusType pg_status = PQresultStatus(result);
-        if (pg_status != PGRES_COMMAND_OK) {
-          AdbcStatusCode code = SetError(
-              error, result, "[libpq] Failed to execute prepared statement: %s %s",
-              PQresStatus(pg_status), PQerrorMessage(conn));
-          PQclear(result);
-          return code;
+          const uint32_t net_nbytes = ToNetworkInt32(static_cast<uint32_t>(field_nbytes));
+          ArrowBufferAppend(&buf, &net_nbytes, sizeof(net_nbytes));
+          ArrowBufferAppend(&buf, param_values[col], field_nbytes);
         }
+      }
 
+      if (PQputCopyData(conn, reinterpret_cast<char*>(buf.data), buf.size_bytes) <= 0) {
```

Comment:
I am not sure if the reinterpret_cast here is appropriate, or if there was a way to construct an ArrowBufferView from the ArrowBuffer and use the union approach